packages/graphback-runtime/src/data/KnexDBDataProvider.ts

```diff
@@ -20,29 +20,37 @@ export class KnexDBDataProvider<Type = any, GraphbackContext = any> implements G
   protected db: Knex;
   protected baseType: GraphQLObjectType;
   protected tableName: string;
+  protected tableMap: ModelTableMap;
 
   constructor(baseType: GraphQLObjectType, db: Knex) {
     this.db = db;
     this.baseType = baseType;
     // TODO build and use mapping here
-    this.tableName = lowerCaseFirstChar(baseType.name);
+    this.tableMap = buildModelTableMap(baseType);
+    this.tableName = this.tableMap.tableName;
   }
 
   public async create(data: Type): Promise<Type> {
-    // TODO refactor to use new knex api
-    const [id] = await this.db(this.tableName).insert(data);
-    const dbResult = await this.db.select().from(this.tableName).where('id', '=', id)
+    const { data: createData } = getDatabaseArguments(this.tableMap, data);
+
+    // tslint:disable-next-line: await-promise
+    const [id] = await this.db(this.tableName).insert(createData);
+    // tslint:disable-next-line: await-promise
+    const dbResult = await this.db.select().from(this.tableName).where(this.tableMap.idField, '=', id)
     if (dbResult && dbResult[0]) {
       return dbResult[0]
     }
     throw new NoDataError(`Cannot create ${this.tableName}`);
   }
 
   public async update(data: Type): Promise<Type> {
-    // TODO refactor to use id mapping
-    const updateResult = await this.db(this.tableName).update(data).where('id', '=', (data as any).id);
+    const { idField, data: updateData } = getDatabaseArguments(this.tableMap, data);
```

Comment:
Lack of validation when `id` is missing

Reply:
Resolved in https://github.com/aerogear/graphback/pull/711/commits/06e6bde642b6c785a5eb9f9890447b9c3ebb3558